### DISCUSSION TITLE
Support for ABlog

### DISCUSSION THREAD

**Discussion by damian-krawczyk** [pradyunsg/furo#262] [Ideas]
**Is your feature request related to a problem? Please describe.**
n/a

**Describe the solution you'd like**
I would like furo theme to support [ABlog](https://ablog.readthedocs.io). It would be good to support even the simplest form: 
 - properly formated list of posts on single page
 - properly formated single article

**Describe alternatives you've considered**
n/a

**Additional context**
See examples of ABlog for Sphinx in use:

- https://www.ericholscher.com
- https://predictablynoisy.com/blog/

**Comment by pradyunsg** [comment]
You'll need to be more specific about what supporting ABlog would mean/need. What works right now, and what doesn't; why Furo needs to be the one that adds compatibility (rather than the other way round etc).

**Comment by pradyunsg** [comment]
Closing due to the lack of a response. Feel free to poke this again.

**Comment by damian-krawczyk** [comment]
@pradyunsg [here](https://github.com/sunpy/ablog/issues/108) I put more details with examples you can check how it works with default theme `alabaster`, `pydata_sphinx_theme` and error for `furo` theme.

**Comment by sudopluto** [comment]
https://github.com/pradyunsg/furo/blob/121edb3e9ade5e7cfc8c75d64190b66d0e7ddb08/docs/changelog.md#20210320beta31

is this relevant?

**Comment by sudopluto** [comment]
looks like (only?) two files to change: https://github.com/sunpy/ablog/search?q=layout.html

**Comment by errbufferoverfl** [comment]
@damian-krawczyk I recently made some changes to the [furo theme to support ablog](https://github.com/errbufferoverfl/furo), as it currently stands it should work as both a normal Sphinx theme, and an ablog theme and did my best to ensure no breaking changes.

If you plan on testing it locally you'll need to follow the instructions on the [workflow](https://pradyunsg.me/furo/contributing/workflow/) to get it to get all the Sass stylesheets and Javascript code working.

Good luck!

**Comment by damian-krawczyk** [comment]
At ablog's end Issue is tracked here https://github.com/sunpy/ablog/issues/108

**Comment by stonecharioteer** [comment]
I just tried to use furo with ablog and got this error in the browser.

```
ERROR: This page was generated by inheriting from "layout.html".

Furo (the documentation theme that this site is generated with) does not support this, and has generated this page. If you are the author of this documentation, please look into why this page was generated using the "layout.html" template and fix the root cause.
Debugging information, if you need to report this error somewhere:

Sphinx: 4.3.0
Furo: 2021.11.16

pagename: index
```

I'm uncertain if the change needs to happen at the furo end, the ablog end or if it needs to be done at the user end.
Any thoughts? @pradyunsg

**Reply by pradyunsg** [reply]
Furo doesn't use `layout.html` for anything. The "error" you're seeing comes from https://github.com/pradyunsg/furo/blob/main/src/furo/theme/furo/layout.html.

I think there is an assumption being made in ABlog, that every Sphinx theme will define the theme's scaffolding as `layout.html`, which is incorrect in this case.

**Reply by pradyunsg** [reply]
Looking more closely, ABlog is also making significant non-portable assumptions about how Sphinx themes would handle their `block`s in Jinja, which is likely a part of the problem here.

**Comment by ABD-01** [comment]
# Ablog with Furo

## TL;DR
### See [**ablog_with_furo**](https://abd-01.github.io/ablog_with_furo/)

## Body

Hi
I was facing a similar issue while using Ablog along with furo.

Based on comments from 
* sunpy/ablog/issues/108
* #262
* stackoverflow, etc...

I forked Ablog and created [ablog_with_furo](https://github.com/ABD-01/ablog_with_furo)

Now to use this I suggest using the `exts` directory as suggested by [sphinx in their documentation](https://www.sphinx-doc.org/en/master/usage/extensions/index.html#third-party-extensions)

You can use submodule for this 
```sh
git submodule add https://github.com/ABD-01/ablog_with_furo.git exts/ablog
```
For instance how I used it in my [repo](https://github.com/ABD-01/ABD-01.github.io/tree/sphinx/exts)

I have also [added meta data](https://github.com/sunpy/ablog/compare/main...ABD-01:ablog_with_furo:ablog_with_furo?expand=1#diff-76ee6585b2f5e5750b2cb6a97d3bdac85dd70125e16dcd596ed062bc9981409bR436-R560) for posts below the title
![image](https://github.com/user-attachments/assets/27aa985b-1fed-487a-80e0-91af9d5909f8)

You can view the Ablog's documentation build using furo [here](https://abd-01.github.io/ablog_with_furo/)
